service/src/test/java/uk/gov/hmcts/reform/fpl/service/OthersServiceTest.java

```diff
@@ -77,4 +84,109 @@ void shouldBuildExpectedLabelWhenEmptyOthers() {
 
         assertThat(result).isEqualTo("No others on the case");
     }
+
+    @Test
+    void shouldReturnEmptyOthersWhenNoOthersInCaseData() {
+        CaseData caseData = CaseData.builder().build();
+
+        Others others = service.prepareOthers(caseData);
+
+        assertThat(others).isEqualTo(Others.builder().additionalOthers(emptyList()).build());
+    }
+
+    @Test
+    void shouldReturnOthersWhenOthersIsPrePopulated() {
+        List<Element<Other>> additionalOthers = othersWithRemovedConfidentialFields();
+
+        CaseData caseData = buildCaseDataWithOthers(otherWithDetailsHiddenValue("No"), additionalOthers, null);
+
+        Others others = service.prepareOthers(caseData);
+
+        assertThat(caseData.getOthers()).isEqualTo(others);
+    }
+
+    @Test
+    void shouldPrepareOthersWithConfidentialValuesWhenConfidentialOthersIsNotEmpty() {
+        List<Element<Other>> additionalOthersList = othersWithConfidentialFields(randomUUID());
+
+        Other firstOther = othersWithRemovedConfidentialFields().get(0).getValue();
+        List<Element<Other>> confidentialOthers = othersWithConfidentialFields(ID);
+
+        CaseData caseData = buildCaseDataWithOthers(firstOther, additionalOthersList, confidentialOthers);
+
+        Others others = service.prepareOthers(caseData);
+
+        assertThat(others.getFirstOther()).isEqualTo(confidentialOthers.get(0).getValue());
+    }
+
+    @Test
+    void shouldReturnOtherWithoutConfidentialDetailsWhenThereIsNoMatchingConfidentialOther() {
+        Other firstOther = othersWithRemovedConfidentialFields().get(0).getValue();
+        List<Element<Other>> additionalOther = othersWithRemovedConfidentialFields();
+        List<Element<Other>> confidentialOther = othersWithConfidentialFields(randomUUID());
+
+        CaseData caseData = buildCaseDataWithOthers(firstOther, additionalOther, confidentialOther);
+
+        Others others = service.prepareOthers(caseData);
+
+        assertThat(others.getAdditionalOthers()).containsOnly(additionalOther.get(0));
+    }
+
+    @Test
+    void shouldAddExpectedRespondentWhenHiddenDetailsMarkedAsNo() {
+        Other firstOther = otherWithDetailsHiddenValue("No");
```

Comment:
Not sure what this method is trying to do and name doesn't reflect it either

Reply:
On second review functionality looks correct but name should be shouldAddExpectedOtherWhenHiddenDetailsMarkedAsNo

Reply:
done